core/services/workflows/engine.go

```diff
@@ -1297,9 +1298,15 @@ func NewEngine(ctx context.Context, cfg Config) (engine *Engine, err error) {
 
 	workflow.id = cfg.WorkflowID
 	workflow.owner = cfg.WorkflowOwner
-	workflow.hexName = hex.EncodeToString([]byte(cfg.WorkflowName))
 	workflow.name = cfg.WorkflowName
 
+	if cfg.WorkflowNameTransform != nil {
```

Comment:
_nit_ unncessary `else`.  can leave line 1300 and overwrite `workflow.hexName` if the transform exists.